### PR TITLE
ENH: Add debug_ddp flag

### PR DESCRIPTION
**Pull request by kenza-bouzid**
Add debug_ddp to assign different options to `TORCH_DISTRIBUTED_DEBUG` env variable.

### CODE REVIEW

File: hi-ml/src/health_ml/runner.py [hunk 1]
```diff
@@ -216,6 +216,7 @@ def after_submission_hook(azure_run: Run) -> None:
 
         # TODO: Update environment variables
         environment_variables: Dict[str, Any] = {}
+        environment_variables[DEBUG_DDP_ENV_VAR] = self.experiment_config.debug_ddp.value  # type: ignore
```
Comment (kenza-bouzid) [inline hi-ml/src/health_ml/runner.py:219]: I had to ignore typin cause mypy was failing with `"str" has no attribute "value" ` error but debug_ddp is an enum any suggestions?

Comment (ant0nsc) [reply hi-ml/src/health_ml/runner.py:219]: The problem is in line 40 of `experiment_config.py`. You are marking the flag as `str`, but it should be `DebugDDPOptions`

File: hi-ml/src/health_ml/experiment_config.py [hunk 2]
```diff
@@ -27,3 +37,10 @@ class ExperimentConfig(param.Parameterized):
                             doc="The Conda environment file that should be used when submitting the present run to "
                                 "AzureML. If not specified, the environment file in the current folder or one of its "
                                 "parents will be used.")
+    debug_ddp: str = param.ClassSelector(default=DebugDDPOptions.OFF, class_=DebugDDPOptions,
```
Comment (ant0nsc) [inline hi-ml/src/health_ml/experiment_config.py:40]: for the typing error below:
```suggestion
    debug_ddp: DebugDDPOptions = param.ClassSelector(default=DebugDDPOptions.OFF, class_=DebugDDPOptions,
```

**Review by ant0nsc — APPROVED** [review]
Looking good - I added a suggestion to your typing question.

**Review by mebristo — APPROVED** [review]

File: hi-ml/src/health_ml/runner.py (hunk 1 above)
Comment (kenza-bouzid) [reply hi-ml/src/health_ml/runner.py:219]: Yees, thanks :)

**Pull request merged**